async-query-core/src/main/java/org/opensearch/sql/spark/validator/S3GlueGrammarElementValidator.java

```diff
@@ -50,6 +55,11 @@ public class S3GlueGrammarElementValidator extends DenyListGrammarElementValidat
               HINTS,
               INLINE_TABLE,
               FILE,
+              CROSS_JOIN,
+              LEFT_SEMI_JOIN,
+              RIGHT_OUTER_JOIN,
+              FULL_OUTER_JOIN,
+              LEFT_ANTI_JOIN,
```

Comment:
These JOIN types are disallowed even for S3 data source?

Reply:
Yes, limiting regardless of the data source.

Reply:
So we only limit these JOIN types due to performance/cost? Others like INNER JOIN, LEFT JOIN can still work?

Reply:
Yes, intention is only allowing INNER JOIN and LEFT OUTER JOIN.